File: PostmanTest/StampTests/SubjectStampTest.vb
Imports System.Net.Mail
Imports Postman

''' <summary>
''' This is a test class for SubjectStamp and is intended to contain all SubjectStamp Unit Tests
''' </summary>
<TestClass()> _
Public Class SubjectStampTest

    '''<summary>
    '''A test for SubjectStamp Constructor with a string containing a subject
    '''</summary>
    <TestMethod()> _
    Public Sub AttachTest()
        'Arrange
        Dim expectedSubject As String = "expectedSubject"
        Dim msg As New MailMessage()
        Dim target As IStamp = New Stamp.Subject(expectedSubject)

        'Act
        target.Attach(msg)

        'Assert
        Assert.AreEqual(expectedSubject, msg.Subject)
    End Sub

End Class

File: Postman.Tests/Stamp/SubjectStampTest.cs
namespace Postman.Tests
{
    using System.Net.Mail;
    using Postman.Interfaces;
    using Postman.Stamp;
    using Xunit;

    /// <summary>
    /// This is a test class for SubjectStamp and is intended to contain all SubjectStamp Unit Tests
    /// </summary>
    public class SubjectStampTest
    {
        /// <summary>
        /// A test for SubjectStamp Constructor with a string containing a subject
        /// </summary>
        [Fact]
        public void AttachTest()
        {
            // Arrange
            string expectedSubject = "expectedSubject";
            MailMessage msg = new MailMessage();
            IStamp target = new Subject(expectedSubject);

            // Act
            target.Attach(msg);

            // Assert
            Assert.Equal(expectedSubject, msg.Subject);
        }
    }
}